### DISCUSSION TITLE
Issue with Dapps Using WalletConnect Integration

### DISCUSSION THREAD

**Discussion by heithemkacem** [WalletConnect/walletconnect-monorepo#3667] [Developer Support] [answered]
Dear WalletConnect Support Team,

I hope this message finds you well. We've recently added our platform to the WalletConnect explorer, and we've encountered an issue with some Dapps not working as expected. Instead of connecting, they redirect users to this URL: [https://www.kayros.games/wc/?uri=wc%3A96914c4526ec36f2d2116732ce4d649e989a16e6fb511e6b29d08c71a674dc9c%402%3Frelay-protocol%3Dirn%26symKey%3D4a939b9836696f70a56c23f69d8d8ff6df6c4dafc315b044b29fcced9d2e78e1.](url)

We suspect that these Dapps might still be using the deprecated WalletConnect v1 integration. Could you confirm if this is the case, and if so, provide guidance on how to handle these instances? Additionally, if there are other steps or best practices to address this issue, we'd appreciate your insights.

We've also noticed the repetition of "kayros" in some redirection URLs, like https://stake.deelance.com/. Could you explain why this occurs and whether it affects WalletConnect's functionality with these Dapps?

Ensuring a seamless WalletConnect integration is crucial for our platform, and we're committed to resolving these issues promptly. Your assistance would be invaluable.

Thank you for your attention, and we look forward to your guidance in resolving these matters.

Best regards,

**Comment by glitch-txs** [comment]
Hi what type of wallet is it?

**Reply by heithemkacem** [reply]
Injected wallet

**Reply by glitch-txs** [reply]
Could you check in your form if webapp url was added and remove it if so?

**Reply by heithemkacem** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/86875191/38787cfa-872a-4522-b19d-c635ca760e73)

**Reply by heithemkacem** [reply]
@glitch-txs submit it and once it get accepted i will reverify the tests and if it works i will mark it as answer

**Reply by heithemkacem** [reply]
@glitch-txs  i have retested same problem 
here some websites examples : 
https://decentral.games/ice
https://opensea.io/
https://lootex.io/

**Reply by heithemkacem** [reply]
@glitch-txs here some other websites that works
https://derbystars.com/market/
https://stake.deelance.com/

**Reply by glitch-txs** [reply, marked as answer]
first two website are using a WalletConnect package that doesn't work with extension wallets in general, last one is using v1, needs to upgrade to v2.

**Reply by glitch-txs** [reply]
I would highly recommend integrating [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) since some modal libraries will start implementing it (including us on Web3Modal v3 and wagmi)
